src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -0,0 +1,214 @@
+                    var latestAutomaticAPIRevision = automaticRevisions.Peek();
+                    var comments = await _commentsManager.GetCommentsAsync(review.Id);
+
+                    while (
+                        automaticRevisions.Any() &&
+                        !latestAutomaticAPIRevision.IsApproved &&
+                        !await _apiRevisionsManager.IsAPIRevisionTheSame(latestAutomaticAPIRevision, renderedCodeFile) &&
+                        !comments.Any(c => latestAutomaticAPIRevision.Id == c.RevisionId))
+                    {
+                        await _apiRevisionsManager.SoftDeleteAPIRevisionAsync(user: User, apiRevision: latestAutomaticAPIRevision);
+                        latestAutomaticAPIRevision = automaticRevisions.Dequeue();
+                    }
+
+                    // We should compare against only latest revision when calling this API from scheduled CI runs
+                    // But any manual pipeline run at release time should compare against all approved revisions to ensure hotfix release doesn't have API change
+                    // If review surface doesn't match with any approved revisions then we will create new revision if it doesn't match pending latest revision
+
+                    if (compareAllRevisions)
+                    {
+                        foreach (var approvedAPIRevision in apiRevisions.Where(r => r.IsApproved))
+                        {
+                            if (await _apiRevisionsManager.IsAPIRevisionTheSame(approvedAPIRevision, renderedCodeFile))
+                            {
+                                return approvedAPIRevision;
+                            }
+                        }
+                    }
+
+                    if (await _apiRevisionsManager.IsAPIRevisionTheSame(latestAutomaticAPIRevision, renderedCodeFile))
+                    {
+                        apiRevision = latestAutomaticAPIRevision;
+                        createNewRevision = false;
+                    }
+                }
+            }
+            else
+            {
+                review = await _reviewManager.CreateReviewAsync(packageName: codeFile.PackageName, language: codeFile.Language, isClosed: false);
+            }
+
+            // Check if user is authorized to modify automatic review
+            await ManagerHelpers.AssertAutomaticReviewModifier(user: User, review: review, authorizationService: _authorizationService);
+            if (createNewRevision)
+            {
+                apiRevision = await _apiRevisionsManager.CreateAPIRevisionAsync(user: User, reviewId: review.Id, apiRevisionType: APIRevisionType.Automatic, label: label, memoryStream: memoryStream, codeFile: codeFile, originalName: originalName);
+            }
+
+            if (apiRevision != null)
+            {
+                if (!apiRevision.IsApproved)
+                {
+                    var latestApprovedAPIRevision = apiRevisions.FirstOrDefault(r => r.IsApproved);
```

Comment:
We need to check against all approved revisions. It is possible to have the latest approved revision is created from a different release branch but the API surface in new request matches with previously approved revision. This happens when a team releases a hotfix or a branched out release branch.

Reply:
Updated to check all apiRevisions